onnx/version_converter/adapters/broadcast_backward_compatibility.h

```diff
@@ -0,0 +1,75 @@
+// Adapter for broadcasting ops in default domain from version 7 to 6
+
+#pragma once
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class BroadcastBackwardCompatibility final : public Adapter {
+  public:
+    explicit BroadcastBackwardCompatibility(const std::string& op_name, const OpSetID
+      initial, const OpSetID target): Adapter(op_name, std::move(
+        initial), std::move(target)) {}
+
+    void adapt_broadcast_backward_compatibility(std::shared_ptr<Graph> graph, Node* node) const {
+      // Verify that broadcasts are allowed in limited spec of opset version 6
+      // Multidirectional broadcasting, as defined in Broadcasting.md
+      // MathDocGenerator provides differences
+      // Main change: encode broadcasting commands as explicit attribute
+      // TODO: Check on this reference.
+      const ArrayRef<Value*>& inputs = node->inputs();
+      ONNX_ASSERTM(inputs.size() == 2, "Add in opset version 6 can only broadcast"
+        " between 2 inputs");
+      std::vector<Dimension> A_sizes = inputs[0]->sizes();
```

Comment:
Check has_sizes before you use the real size: https://github.com/onnx/onnx/blob/master/onnx/common/ir.h#L297